The Straits Times.js

```diff
@@ -0,0 +1,286 @@
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.match(/(advertise-with-the-straits-times|newsletter-signup)/)) {
+		return 'webpage';
+	}
+	var pageClass = ZU.xpathText(doc, '//meta[@name="cXenseParse:pageclass"]/@content');
+	if (pageClass === 'article') {
+		return 'newspaperArticle';
+	}
+	if (pageClass === 'frontpage') {
+		return 'multiple';
+	}
+	
+	return false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) === "multiple") {
+		getMultipleItems(doc, url);
+	} else {
+		scrape(doc, url, detectWeb(doc, url));
+	}
+}
+
+function scrape(doc, url, type) {
+	var newItem = new Zotero.Item(type || 'newspaperArticle');
+	newItem.ISSN = '0585-3923';
+	newItem.url = url;
+	newItem.publicationTitle = 'The Straits Times';
+	newItem.title = ZU.xpathText(doc, '//meta[@name="dcterms.title"]/@content');
+	newItem.abstractNote = ZU.xpathText(doc, '//meta[@name="dcterms.description"]/@content') ? ZU.xpathText(doc, '//meta[@name="dcterms.description"]/@content').replace('Read more at straitstimes.com.', '').trim() : '';
+	newItem.date = ZU.xpathText(doc, '//meta[@name="dcterms.date"]/@content');
+	newItem.place = 'Singapore';
+	newItem.language = 'en';
+	var authors = ZU.xpath(doc, '//meta[@property="article:author"]');
+	for (var i = 0; i < authors.length; i++) {
+		insertCreator(authors[i].getAttribute('content'), newItem);
+	}
+	newItem.attachments = [{
+		document: doc,
+		title: "The Straits Times Snapshot",
+	}];
+	if (doc.evaluate('//div[@class="paid-premium st-flag-1"]', doc, null, XPathResult.ANY_TYPE, null).iterateNext()) {
+		newItem.extra = "(Subscription only)";
+	}
+	newItem.complete();
+}
+
+function getMultipleItems(doc, url) {
+	var items = [];
+	var rows;
+	if (url.includes('/search?') && url.includes('searchKey')) {
+		rows = ZU.xpath(doc, '//div[@class="queryly_item_row"]');
+		if (rows.length) {
+			for (var i = 0; i < rows.length; i++) {
+				var searchItem = rows[i];
+				var searchItemUrl = searchItem.querySelector('a') ? searchItem.querySelector('a').href : '';
+				items.push(searchItemUrl);
+			}
+		}
+	} else {
+		rows = ZU.xpath(doc, '//a[@class="block-link"]|//span[@class="story-headline"]/a');
+		if (rows.length) {
+			for (var k = 0; k < rows.length; k++) {
+				var headlineItem = rows[k];
+				var headlineItemUrl = headlineItem.href ? headlineItem.href : '';
+				items.push(headlineItemUrl);
+			}
+		}
+	}
+	if (!!items && items.length) {
+		items = items.filter(function(item) {
+			return (!!item.match(/^https:\/\/www\.straitstimes.com/));
+		});
+		if (items.length) {
+			ZU.processDocuments(items, scrape);
+		}
+	}
+}
+
+function insertCreator(authorName, newItem) {
+	// to account for mostly Chinese names in formats of: <last> <first>, <first (typically English)> <last> <first (typically Chinese)>
```

Comment:
It's really unfortunate that they don't give us a better way to figure out name order in the metadata! It feels wrong to have to embed a list of authors that will become out of date over time. But it might be our only option.

Reply:
yeah, unfortunately. This is an artefact of Singaporean society. After growing up here, we would know the general race of a person from the name, and which is the first name and which is the surname/family/patronymic name. I couldn't think of a better way unless we forgo breaking down into first and last name.